core/network.kt

```diff
@@ -0,0 +1,46 @@
+private fun isInternetConnected(context: Context): Boolean {
+    var result = false
+    val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
```

Comment:
might be an android kotlin extension library but theres nice kotlin helpers that make this really easy:
`context.getSystemService<ConnectivityManager>()`
also technically this is nullable